File: C/1. Basic Concepts/4.c
// Operators
#include <stdio.h>

int main()
{
    int f = 5, s = 2;
    printf("5 + 2 = %d\n", f + s);
    printf("5 - 2 = %d\n", f - s);
    printf("5 * 2 = %d\n", f * s);
    printf("5 / 2 = %d\n", f / s);
    printf("5 %% 2 = %d\n", f % s);

    int x = 23, y = 4;

    // Type casting
    float sum = (float)x / y; // only this ay can get float after divide int on int
    printf("%f\n", sum);

    // Assignment Operators
    x += y;
    x -= y;
    x *= y;
    x /= y;
    x %= y;

    // Increment & Decrement
    /*
        prefix (--z) increments/decrements the variable and then uses it in the assignment statement
        postfix (++z) uses the value of the variable first, before incrementing/decrementing it
    */
    int z = 10;
    z--;
    printf("%d\n", z);
    z++;
    printf("%d\n\n", z);

    z = 3;
    x = z--; // assign 3 to x, then decrement z to 2
    printf("x=%d z=%d \n", x, z);

    y = 3;
    x = ++y; // increment y to 4, then assign 4 to x
    printf("x=%d y=%d\n", x, y);

    return 0;
}

File: Java/1. Basic Concepts/4.java
class Operators {
    public static void main(String[] args) {
        // Primitive Operators
        int x = 7, y = 5;
        int i1, i2, i3, i4, i5;

        i1 = x + y; // 12
        i2 = x - y; // 2
        i3 = x * y; // 35
        i4 = x / y; // 1
        i5 = x % y; // 2

        System.out.print("i1 = 7 + 5 = ");
        System.out.println(i1);
        System.out.print("i2 = 7 - 5 = ");
        System.out.println(i2);
        System.out.print("i3 = 7 * 5 = ");
        System.out.println(i3);
        System.out.print("i4 = 7 / 5 = ");
        System.out.println(i4);
        System.out.print("i5 = 7 % 5 = ");
        System.out.println(i5);
        System.out.println();

        // Increment & Decrement
        int z = 14;
        System.out.println(z++); // Prints z (14) and increments to 15
        System.out.println(++z); // Increments z to 16 and prints it
        System.out.println();

        // Assignment and operation
        z += 2; // 18
        z -= 1; // 17
        z *= 2; // 34
        z /= 3; // 11
        z %= 4; // 3
        System.out.println(z);
    }
}
